release_checklist.rb

```diff
@@ -0,0 +1,218 @@
+require 'tempfile'
+
+def use_local_template?
+  ENV['GBM_USE_LOCAL_RELEASE_TEMPLATES']
+end
+
+# Output helpers
+def say(message)
+  STDERR.print(message)
+end
+
+def exit_with(message)
+  say(message)
+  exit
+end
+
+def abort_with(message)
+  say(message)
+  exit(1)
+end
+
+class Checklist < ERB
+  def task_block (description)
+  %Q(
+  <!-- wp:p2/task {"assigneesList":[]} -->
+    <div class="wp-block-p2-task">
+      <div>
+        <span class="wp-block-p2-task__emoji-status" title="Pending">⬜ </span>
+        <div class="wp-block-p2-task__checkbox-wrapper">
+          <span title="Pending" class="wp-block-p2-task__checkbox is-disabled is-aria-checked-false"></span>
+        </div>
+      </div>
+      <div class="wp-block-p2-task__main">
+        <div class="wp-block-p2-task__left">
+          <div class="wp-block-p2-task__content-wrapper">
+            <span class="wp-block-p2-task__content">#{description}</span>
+          </div>
+          <div class="wp-block-p2-task__dates"></div>
+        </div>
+        <div class="wp-block-p2-task__right">
+          <div class="wp-block-p2-task__assignees-avatars"></div>
+        </div>
+      </div>
+    </div>
+  <!-- /wp:p2/task -->
+  )
+  end
+end
+
+
+class ReleaseChecklist < Checklist
+
+  def self.template
+    if use_local_template?
+      return File.read('./templates/release_checklist.html.erb')
+    end
+    open('https://raw.githubusercontent.com/wordpress-mobile/release-toolkit-gutenberg-mobile/trunk/templates/release_checklist.html.erb').read
+  end
+
+  def initialize(version, options = {})
+    @version = version
+    local_template = options.fetch(:template, false)
+    @template = self.class.template
+    @release_date = options.fetch(:release_date, "[date]")
+    @mobile_version = options.fetch(:mobile_version, "XX.X")
+    @is_unscheduled = options.fetch(:is_unscheduled, false)
+    @unscheduled_message = options.fetch(:unscheduled_message, nil)
+    @new_release_url="https://github.com/wordpress-mobile/gutenberg-mobile/releases/new?tag=v#{ @version }&amp;target=release/#{ @version }&amp;title=Release%20<% @version %>"
+
+    @aztec_checklist = options.fetch(:aztec_checklist, '')
+    @incoming_change_checklist = options.fetch(:incoming_change_checklist, '')
+
+    super(@template)
+  end
+
+  def result
+    super(binding)
+  end
+
+end
+
+class AztecChecklist < Checklist
+  def self.template
+    if use_local_template?
+      return File.read('./templates/aztec_checklist.html.erb')
+    end
+    begin 
+      open('https://raw.githubusercontent.com/wordpress-mobile/release-toolkit-gutenberg-mobile/trunk/templates/aztec_checklist.html.erb').read
+    rescue OpenURI::HTTPError => ex 
+      abort_with "Error: #{ex.message}"
+    end
+  end
+
+  def initialize()
+    @template = self.class.template
+    super(@template)
+  end
+
+  def result
+    super(binding)
+  end
+
+  def self.render()
+    new().result
+  end
+end
+
+class IncomingChangeChecklist < Checklist
```

Comment:
Similar to the Aztec checklist, I separated out the incoming changes so that we can only include these steps when needed. In this case we could update the checklist generator script to output just the "incoming changes" and then the release wrangler appends the block of steps to the existing checklist.

I should also mention that the release window has narrowed so that I'm not entirely sure we need these steps at all. We now prepare and cut the release the same day. Even if we need to make a last minute change I not sure it warrants a full set of checklist steps.